core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1810,7 +1811,7 @@ && checkForStartOffsetWithinBatch(inFlightBatch.firstOffset(), inFlightBatch.las
                     // If we have an acquisition lock timeout for a share-partition, then we should check if
                     // there is a pending share fetch request for the share-partition and complete it.
                     DelayedShareFetchKey delayedShareFetchKey = new DelayedShareFetchGroupKey(groupId, topicIdPartition.topicId(), topicIdPartition.partition());
-                    delayedShareFetchPurgatory.checkAndComplete(delayedShareFetchKey);
+                    replicaManager.completeDelayedShareFetchRequest(delayedShareFetchKey);
```

Comment:
If there happens to be a function/callback defined in SharePartitionManager which accepts `delayedShareFetchKey` then that method can be re-used at other places as well where we want to invoke replica manager API calls.

I would suggest the least dependency of scala classes in other classes so we can move them from core to share module.

Reply:
@apoorvmittal10 , we've added `replicaManager.completeDelayedShareFetchRequest(delayedShareFetchKey)` on acqusiition lock timeout functionality. This is independent of `SharePartitionManager` functionalities like acknowledge, release acquired records on session close etc. Hence, if we're not passing `ReplicaManager` object, we'll have to pass `SharePartitionManager` object to `SharePartition` class if we have to steer from `ReplicaManager`. Let me know if you have any other alternatives.